src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -334,16 +334,53 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     * @param requestContext RequestContext
     * @param userInfo       UserInfo
     * @param spreadsheetId  Spreadsheet ID
-    * @param content        ValueRange
+    * @param newContent     ValueRange
     * @return               JsObject representing the Google Update response
     */
-  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, content: ValueRange): JsObject = {
+  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, newContent: ValueRange): JsObject = {
     val credential = new GoogleCredential().setAccessToken(userInfo.accessToken.token)
     val service = new Sheets.Builder(httpTransport, jsonFactory, credential).setApplicationName("FireCloud").build()
-    val response = service.spreadsheets().values().update(spreadsheetId, content.getRange, content).setValueInputOption("RAW").execute()
+
+    // Retrieve existing records
+    val getExisting: Sheets#Spreadsheets#Values#Get = service.spreadsheets().values.get(spreadsheetId, "Sheet1")
+    val existingContent: ValueRange = getExisting.execute()
+
+    // Smart merge existing with new
+    val rows = updatePreservingOrder(newContent, existingContent)
+
+    // Send update
+    val response = service.spreadsheets().values().update(spreadsheetId, newContent.getRange, newContent.setValues(rows)).setValueInputOption("RAW").execute()
     response.toString.parseJson.asJsObject
   }
 
+  private def updatePreservingOrder(newContent: ValueRange, existingContent: ValueRange): List[java.util.List[AnyRef]] = {
+    val header: java.util.List[AnyRef] = existingContent.getValues.get(0)
+    val existingRecords: List[java.util.List[AnyRef]] = existingContent.getValues.drop(1).toList
+
+    val newRecords: List[java.util.List[AnyRef]] = newContent.getValues.drop(1).toList
+
+    // Go through existing records and update them in place
+    val existingRecordsUpdated = existingRecords.map { existingRecord =>
+      val matchingNewRecord = newRecords.find { newRecordCandidate =>
+        newRecordCandidate.get(0) == existingRecord.get(0)
+      }
+
+      matchingNewRecord match {
+        case Some(newRecord) => newRecord // Overwrite the entire row in place, adequate unless we want to preserve user-added columns
```

Comment:
I do think we should consider preserving user-added columns, not necessarily in this PR. One of the reasons for keeping a data row in the same order would be this exact case - the user has added a new column with a comment that applies to the current row. If we are not trying to preserve user-comments in additional columns, then there's no real point in preserving the order of the data rows. 

That being said, I think this might still work (will test functionally). A new `ValueRange`, when pushed to the spreadsheet, *only* overwrites values *inside* the range and leaves values (and formats, etc.) outside the range intact.

Reply:
Update. I believe this is functioning as it should: 
* New records do not overwrite data in user-added columns and instead are appended as new rows. 
* Existing records are overwritten in place without changing user-added columns outside of the updated data. The caveat to this is that if new columns are added to the output beyond what the existing spreadsheet has, they *will* overwrite the user-added columns that exist in the same position.